src/Shared/CopyOnWriteDictionary.cs

```diff
@@ -548,7 +550,9 @@ internal bool HasSameBacking(CopyOnWriteDictionary<K, V> other)
         /// </summary>
         /// <typeparam name="K1">The key type.</typeparam>
         /// <typeparam name="V1">The value type.</typeparam>
+#if FEATURE_BINARY_SERIALIZATION
```

Comment:
There's a [SerializableAttribute](https://github.com/Microsoft/msbuild/blob/31b8984e99b6a55bb0fad4e16582f06ece4605eb/src/Shared/Compat/SerializableAttribute.cs) defined in src/Shared/Compat that you can use so you don't have to put #if's around `[Serializable]` or `[NonSerialized]`.  The pattern I've been using is to include it if the TargetFrameworkIdentifier is not .NETFramework.

``` xml
  <ItemGroup Condition="$(TargetFrameworkIdentifier) != '.NETFramework'">
    <Compile Include="..\Shared\Compat\SerializableAttribute.cs">
      <Link>SharedUtilities\Compat\SerializableAttribute.cs</Link>
    </Compile>
  </ItemGroup>
```